sdk/communication/azure-communication-callautomation/azure/communication/callautomation/aio/_call_connection_client_async.py

```diff
@@ -0,0 +1,553 @@
+        :type call_connection_id: str
+        :return: CallConnectionClient
+        :rtype: ~azure.communication.callautomation.CallConnectionClient
+        """
+        endpoint, access_key = parse_connection_str(conn_str)
+        return cls(endpoint, access_key, call_connection_id, **kwargs)
+
+    @classmethod
+    def _from_callautomation_client(
+        cls,
+        callautomation_client: 'CallAutomationClient',
+        call_connection_id: str
+    ) -> 'CallConnectionClient':
+        """Internal constructor for sharing the pipeline with CallAutomationClient."""
+        return cls(None, None, call_connection_id, _callautomation_client=callautomation_client)
+
+    def get_call_properties(self, **kwargs) -> CallConnectionProperties:
+        """Get the latest properties of the call.
+
+        :return: CallConnectionProperties
+        :rtype: ~azure.communication.callautomation.CallConnectionProperties
+        :raises ~azure.core.exceptions.HttpResponseError
+        """
+
+        call_properties = self._call_connection_client.get_call(call_connection_id=self._call_connection_id, **kwargs)
+
+        return CallConnectionProperties._from_generated(call_properties) # pylint:disable=protected-access
+
+    @distributed_trace_async
+    async def hang_up(self, is_for_everyone: bool, **kwargs) -> None:
+        """Hangup the call.
+
+        :param is_for_everyone: Determine if the call should be ended for all participants.
+        :type is_for_everyone: bool
+        :return: None
+        :rtype: None
+        :raises ~azure.core.exceptions.HttpResponseError
+        """
+
+        if is_for_everyone:
+            await self._call_connection_client.terminate_call(
+                self._call_connection_id,
+                repeatability_first_sent=get_repeatability_guid(),
+                repeatability_request_id=get_repeatability_timestamp(),
+                **kwargs)
+        else:
+            await self._call_connection_client.hangup_call(
+                self._call_connection_id, **kwargs)
+
+    @distributed_trace_async
+    async def get_participant(
+        self,
+        target_participant: 'CommunicationIdentifier',
+        **kwargs
+    ) -> 'CallParticipant':
+        """Get details of a participant in a call.
+
+        :param target_participant: The participant to retrieve.
+        :type target_participant: ~azure.communication.callautomation.CommunicationIdentifier
+        :return: CallParticipant
+        :rtype: ~azure.communication.callautomation.CallParticipant
+        :raises ~azure.core.exceptions.HttpResponseError
+        """
+
+        participant = self._call_connection_client.get_participant(
+            self._call_connection_id, target_participant.raw_id, **kwargs)
+
+        return await CallParticipant._from_generated(participant) # pylint:disable=protected-access
+
+    @distributed_trace_async
+    async def list_participants(self, **kwargs) -> AsyncItemPaged[CallParticipant]:
```

Comment:
This should actually be a sync method, as it returns an async iterable.
```suggestion
    def list_participants(self, **kwargs) -> AsyncItemPaged[CallParticipant]:
```